api/System/SelfUpdate.php

```diff
@@ -273,6 +273,12 @@ private function validate(string $tempFile, string $sha1): void
      */
     private function install(string $tempFile, string $phar): void
     {
+        if ($this->isWindows()) {
```

Comment:
can you please copy the platform check here since we're not using the private method anywhere else?

Reply:
done